### PR TITLE
Rename prop from "state" to "data"

### PR DESCRIPTION
**Pull request by GHaberis**
Renames the `prop` "state" to "data" for clarity purposes.

Our `props` should be used for passing data from parent to child components, and should be  immutable whilst our `state` should be mutable and is used for interactivity to trigger re-renders.

### CODE REVIEW

File: src/lib/__html.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ export default ({
             <script>
             window.gu = {
                 app: {
-                    state: ${JSON.stringify(state)},
+                    app: ${JSON.stringify(data)},
```
Comment (SiAdcock) [inline src/lib/__html.js]: Shall we call this field `data` for consistency?

Comment (sndrs) [reply src/lib/__html.js]: `app.app`?

Comment (GHaberis) [reply src/lib/__html.js]: Yes definitely, that was missed...

**Comment by GHaberis** [comment]
I opted for the name `data` in the end as `app` was already defined on `window.gu`, so we now end up with `window.gu.app.data` rather than `window.gu.app.app`...which we'd have had if I'd used `app`.

**Comment by sndrs** [comment]
will we have anything else on `window.gu.app`?

**Comment by sndrs** [comment]
actually it's fine for now. we need a proper state management thing anyway and this clears up a differnet problem 👍

**Pull request merged**